Fix duplicate param check for ref arrays

bolknote opened the pull request:
Code:
`define a(*t[], t[]) {}`

Expected result:
Parse error: function parameter or auto "t[]" already exists

Actual result:
Parses successfully (incorrect)

The interpreter now correctly rejects functions with duplicate parameter names, even when one is a reference array (*t[]).

bolknote committed:
Fix duplicate param check for ref arrays

gavinhoward reviewed (changes requested):
I'll be upfront: I have accepted this because it passes the tests, and yes, it is a bug.

However, if you run `./scripts/format.sh` on the repo before I do that, I won't have to adjust the style, which would hide your contribution for the `if` statement in `src/lang.c`.

If you could do that, I'd appreciate it, and you would get to keep your credit. If not, no problem; just tell me.

gavinhoward commented:
Oh, and another thing you could get credit for: if you create `tests/bc/errors/39.txt` and put your reproducer into it, I would accept that too.

bolknote committed:
Fix coding style

bolknote committed:
Add test for duplicate ref array parameters

This test ensures the interpreter correctly rejects functions with
duplicate parameter names, even when one is a reference array (*t[]).

bolknote commented:
Yes, no problem! I hope I’ve managed to cover everything needed. If something isn’t quite right, please let me know!

gavinhoward commented:
Looks good!
